parts_server.rb

```diff
@@ -168,7 +168,7 @@ def send_email(to, subject, body)
     end
 
     get "/projects/:id/dashboard/parts" do
-      @status = params[:status] if Part::STATUS_MAP.has_key?(params[:status])
```

Comment:
I get what you're trying to do here, but given that there''s more complication in filtering the statuses now I would refactor it to determine them here instead of in the template. For example:

```ruby
get "/projects/:id/dashboard/parts" do
  if Part::STATUS_MAP.has_key?(params[:status])
    @statuses = [params[:status]]
  else
    @statuses = Part::STATUS_MAP.keys
    @statuses.delete("done") if params[:status] == "progress"
  end
  erb :dashboard_parts
end
```